contributing/data-model-schema.txt

```diff
@@ -0,0 +1,321 @@
+in :file:`ome.xsd`'s first `xsd:documentation` tag. Likewise, with the
+XML transforms, update the copyright date in their headers, and in the
+attributes appearing near the start of :bf_source:`ome-transforms.xml
+<components/specification/transforms/ome-transforms.xml>`.
+
+Other files in which to fix the schema version include:
+
+* :bf_source:`build.properties <components/autogen/build.properties>`
+  and :bf_source:`xsd-fu.xml <ant/xsd-fu.xml>` for code generation
+* the Project Object Model, Maven's :bf_source:`pom.xml`
+* the :bf_source:`publish script <components/specification/publish>`
+  because of the HTML within
+* checks in the Bio-Formats code for the latest schema version,
+  including various Java classes (`version.equals`, `SCHEMA_LOCATION`,
+  etc.)
+
+Avoid changing:
+
+* sample files in :bf_source:`components/specification/samples`
+* documentation in :bf_source:`docs/sphinx`
+* old schema releases
+
+
+Testing
+^^^^^^^
+
+.. TODO:
+   After the 5.1 release and redirect, the link below can become
+   :bf_doc:`C++ implementation <developers/cpp.html>`
+
+Once the above changes have been made and committed, it is time to test.
+This requires having various prerequisites installed for Bio-Formats
+development, including for the `C++ implementation
+<http://www.openmicroscopy.org/site/support/bio-formats5.1/developers/cpp/overview.html>`_.
+Before each test, :ref:`clean the repository <clean_bf_repo>`::
+
+  git clean -dfx
+  ant test
+  git clean -dfx
+  mvn test
+  git clean -dfx
+  TMPDIR=/tmp/bf-build-`date +%s`
+  mkdir $TMPDIR
+  pushd $TMPDIR
+  cmake `dirs +1`
+  make
+  ctest -V
+  popd
+
+You may care to give :command:`make` an additional :option:`-j` option
+specifying the number of cores to use in parallelizing the build. Note
+that the :command:`ctest` step can take a long time.
+
+Sample files
+------------
+
+OME-XML sample files
+^^^^^^^^^^^^^^^^^^^^
+
+Once the schemas and transforms are moved and named to fit the release
+version, then the sample files can be upgraded. A new copy of the sample
+files is created in a new directory, updated to the new schema using
+:command:`xsltproc` with the new transform, then pretty-printed with
+:command:`xmllint` or similar. A sufficient command-line approach is::
+
+  cd components/specification/samples
+  for SRC in `find 2015-01 -type f -name '*.ome' -o -name '*.xml'`
+  do DEST=`echo $SRC | sed -e 's/^2015-01/2016-06/‘`
+     mkdir -p `dirname $DEST`
+     <$SRC xsltproc ../transforms/2015-01-to-2016-06.xsl - | xmllint --format - >$DEST
+  done
+
+The OME-TIFF files require special handling, as they do not have an
+automatic update tool. First, identify them and copy them to the new
+directory::
+
+  find 2015-01 -name '*.ome.tiff'
+  cp 2015-01/set-1-meta-companion/*.ome.tiff 2016-06/set-1-meta-companion/
+
+Next, each OME-TIFF file must be edited to have the schema version
+changed to that of the new release. They are binary files so choice of
```

Comment:
Agreed we need a remedy although I do not think we have discussed/agreed on a solution. I could either see some API in ImageConverter to preserve the original file layout or a consumer project of OME Files scripting the OME-TIFF generation.